Examples/Volume/WebXRVolume/index.js

```diff
@@ -0,0 +1,133 @@
+
+// ----------------------------------------------------------------------------
+// Example code
+// ----------------------------------------------------------------------------
+// Server is not sending the .gz and with the compress header
+// Need to fetch the true file name and uncompress it locally
+// ----------------------------------------------------------------------------
+
+const vtiReader = vtkXMLImageDataReader.newInstance();
+const actor = vtkVolume.newInstance();
+const mapper = vtkVolumeMapper.newInstance();
+mapper.setInputConnection(vtiReader.getOutputPort());
+actor.setMapper(mapper);
+renderer.addVolume(actor);
+
+// create color and opacity transfer functions
+const ctfun = vtkColorTransferFunction.newInstance();
+const ofun = vtkPiecewiseFunction.newInstance();
+
+let fileURL =
+  'https://data.kitware.com/api/v1/file/59de9dca8d777f31ac641dc2/download';
+if (userParams.fileURL) {
+  fileURL = userParams.fileURL;
+}
+
+HttpDataAccessHelper.fetchBinary(fileURL).then((fileContents) => {
+  // Read data
+  vtiReader.parseAsArrayBuffer(fileContents);
+  const data = vtiReader.getOutputData(0);
+  const dataArray =
+    data.getPointData().getScalars() || data.getPointData().getArrays()[0];
+  const dataRange = dataArray.getRange();
+
+  // Restyle visual appearance
+  const sampleDistance =
+    0.7 *
+    Math.sqrt(
+      data
+        .getSpacing()
+        .map((v) => v * v)
+        .reduce((a, b) => a + b, 0)
+    );
+  mapper.setSampleDistance(sampleDistance);
+
+  ctfun.addRGBPoint(dataRange[0], 0.0, 0.3, 0.3);
+  ctfun.addRGBPoint(dataRange[1], 1.0, 1.0, 1.0);
+  ofun.addPoint(dataRange[0], 0.0);
+  ofun.addPoint((dataRange[1] - dataRange[0]) / 4, 0.0);
+  ofun.addPoint(dataRange[1], 0.5);
+  actor.getProperty().setRGBTransferFunction(0, ctfun);
+  actor.getProperty().setScalarOpacity(0, ofun);
+  actor.getProperty().setInterpolationTypeToLinear();
+
+  // Set up rendering
+  const interactor = renderWindow.getInteractor();
+  interactor.setDesiredUpdateRate(15.0);
+  renderer.resetCamera();
+  renderWindow.render();
+
+  // Add button to launch AR (default) or VR scene
+  const VR = 1;
+  const AR = 2;
+  let xrSessionType = 0;
+  const xrButton = document.createElement('button');
+  let buttonContent = 'XR not available!';
+  if (
+    navigator.xr !== undefined &&
+    fullScreenRenderer.getApiSpecificRenderWindow().getXrSupported()
+  ) {
+    if (navigator.xr.isSessionSupported('immersive-ar')) {
+      xrSessionType = AR;
+      buttonContent = 'Start AR';
+    } else if (navigator.xr.isSessionSupported('immersive-vr')) {
+      xrSessionType = VR;
+      buttonContent = 'Start VR';
+    }
+  }
+  xrButton.textContent = buttonContent;
+  xrButton.addEventListener('click', () => {
+    if (xrSessionType === AR) {
+      fullScreenRenderer.setBackground([0, 0, 0, 0]);
+    }
+    fullScreenRenderer
+      .getApiSpecificRenderWindow()
+      .startXR(xrSessionType === AR);
+    xrButton.textContent = 'Exit XR';
```

Comment:
Lies?  (Just calls `startXR` again?)

Reply:
Updated so that `Exit XR` button actually closes the XR session on request.